### PR TITLE
Embrace QGIS' reuse last entered value setting

### PR DESCRIPTION
**Pull request by nirvn**
QGIS has a nice attribute form field configuration that defines whether entered attribute values while digitizing a new feature should be remembered when adding subsequent feature(s):

image

QField has had that option for a long time, in fact longer than when this was introduced, which explains why when the configuration was added upstream, we skipped that beat. This PR embraces the QGIS setting.

This decomplexify data entry for field users as it removes the need to click on that 'remember' checkbox at least once per session, and can help project managers in not having to teach users how to activate this as they can configure it on behalf of users in the project file itself.

The PR also tries to improve the QField UI around this functionality by moving away from a label-less checkbox towards a more descriptive icon tool button. A toaster message was also added to inform users of what is happening when the button is toggled.

Before vs. PR (updated looks with a toast message):
image

Last but not least, we now remember across sessions what fields users want remembered for a given project.

### CODE REVIEW

File: src/qml/FeatureForm.qml [hunk 1]
```diff
@@ -690,7 +690,7 @@ Page {
 
             anchors { right: parent.right; top: constraintDescriptionLabel.bottom; verticalCenter: menuButton.verticalCenter }
 
-            onCheckedChanged: {
+            onClicked: {
```
Comment (suricactus) [inline src/qml/FeatureForm.qml:693]: Why the change here? Just want to learn why so? Because the "changed" one will trigger from the initial qgis values?

**Review by suricactus — APPROVED** [review]
Very good polish, on my personal too list for quite some time. I guess it also worth considering to store the remember checkbox state across QField sessions.

File: src/qml/FeatureForm.qml (hunk 1 above)
Comment (nirvn) [reply src/qml/FeatureForm.qml:696]: @suricactus , yeah, if we used oncheckedchanged, QField (rightfully) complaints about a binding loop.

Comment (nirvn) [reply src/qml/FeatureForm.qml:696]: (and that's because clicking on the toolbar button triggers the oncheckedchanged signal, and in there we modify the RememberValue property which in turn re-triggers an oncheckedchanged signal as the checked property is re-evaluated).

**Comment by nirvn** [comment]
Nice UI tweak bonus: we can a few extra pixels of horizontal space when moving away from checkboxes :party:

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 254ed97ba778be49aafe25f94f30fb87b02ee6a0: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-arm-neon.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-x64.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-x86.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-arm64.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-arm-neon.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-x64.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-5109-4d040118f87aa67bbf1c87b0aa45e17587c3916b-android-x86.apk)
</details>

**Comment by nirvn** [comment]
@suricactus , you stole my punchline :wink: the last part of this set of improvements is indeed remembering across sessions what we decided to remember.

File: src/core/projectinfo.cpp [hunk 2]
```diff
@@ -324,6 +324,42 @@ void ProjectInfo::saveLayerSnappingConfiguration( QgsMapLayer *layer )
...
+  }
+
+  const bool isDataset = QgsProject::instance()->readBoolEntry( QStringLiteral( "QField" ), QStringLiteral( "isDataset" ), false );
+
+  // Prefix id with :: to avoid loss of slash on linux paths
+  QString id( QStringLiteral( "::" ) );
+  if ( isDataset )
+  {
+    // For non-project datasets, the layer id is random, use the source URI
+    id += layer->source();
```
Comment (suricactus) [inline src/core/projectinfo.cpp:351]: I thought when importing a layer as dataset it always had a different path. Good to learn it's not that bad!

**Review by suricactus — APPROVED** [review]
As we discussed in private, the best approach would be to introduce the hamburger menu for each of the menu items and get rid of any kind of icon/checkbox. 

Anyway, this PR provides great improvements of usability of this feature, code looks good. Thanks!
